OSDOCS-9469: Documented the configure-ovs alternative (#76548)
openshift/openshift-docs: modules/creating-scaling-machine-sets-compute-nodes-networking.adoc

// Module included in the following assemblies:
//
// IPI
// * installing/installing_bare_metal_ipi/ipi-install-installation-workflow.adoc
// UPI
// * installing/installing_bare_metal/installing-bare-metal-network-customizations.adoc
// * installing/installing_bare_metal/installing-restricted-networks-bare-metal.adoc
// * installing/installing_bare_metal/installing-bare-metal.adoc

:_mod-docs-content-type: PROCEDURE

> [id="creating-scaling-machine-sets-compute-nodes-networking.adoc_{context}"]

Reviewer: ```suggestion
[id="creating-scaling-machine-sets-compute-nodes-networking_{context}"]
```

Author: :heavy_check_mark:

= Optional: Scaling each machine set to compute nodes

> To apply a customized `br-ex` bridge configuration to all compute nodes in your {product-title} cluster, you must edit your `MachineConfig` custom resource definition (CRD) and modify its roles. Additionally, you must create a `BareMetalHost` CRD that defines information for your bare-metal machine, such as hostname, credentials, and so on.

Reviewer: Just checking that this indeed is about modifying the _CRD_ itself, and not an individual _CR_ instance. The `oc edit mc` command later made me wonder (since that looks more like modifying an individual `Kind` type), but maybe things work different for MachineConfigs. 🤷 

If this should actually be CR, then:
```suggestion
To apply a customized `br-ex` bridge configuration to all compute nodes in your {product-title} cluster, you must edit your `MachineConfig` custom resource (CR) and modify its roles. Additionally, you must create a `BareMetalHost` CR that defines information for your bare-metal machine, such as hostname, credentials, and so on.
```
And also adjust the later instances of "CRD" in this module to "CR" to match.

After you configure these resources, you must scale machine sets, so that the machine sets can apply the resource configuration to each compute node and reboot the nodes.

.Prerequisites

* You created a `MachineConfig` manifest object that includes a customized `br-ex` bridge configuration.

.Procedure

> . Enter the following command to edit the `MachineConfig` CRD:

Reviewer: A couple later steps use the form "Do the thing by entering the following command:", so maybe flip this one around to match:
```suggestion
. Edit the `MachineConfig` CRD by entering the following command:
```

Author: :heavy_check_mark:

+
[source,terminal]
----

> $ oc edit mc <machine_config_CRD_name>

Reviewer: Lowercase for user-replaceables:
```suggestion
$ oc edit mc <machine_config_crd_name>
```
or if this should actually be a CR:
```suggestion
$ oc edit mc <machine_config_cr_name>
```

Author: :heavy_check_mark:

----

. Add each compute node configuration to the CRD, so that the CRD can manage roles for each defined compute node in your cluster.

> . Create an `extraworker-secret` `Secret` object that has a minimal static IP configuration.
> 
> . Apply the `extraworker-secret` `Secret` object to each node in your cluster by entering the following command. This step provides each compute node access to the Ignition config file.

Reviewer: Minor suggest just to avoid back-to-back backtick items:
```suggestion
. Create a `Secret` object named `extraworker-secret` that has a minimal static IP configuration.

. Apply the `extraworker-secret` secret to each node in your cluster by entering the following command. This step provides each compute node access to the Ignition config file.
```

Author: :heavy_check_mark:

+
[source,terminal]
----
$ oc apply -f ./extraworker-secret.yaml
----

> . Create a `BareMetalHost` resource and specify the network secret in the `preprovisioningNetworkDataName` parameter.

Reviewer: ```suggestion
. Create a `BareMetalHost` resource and specify the network secret in the `preprovisioningNetworkDataName` parameter:
```

Author: :heavy_check_mark:

+
.Example `BareMetalHost` resource with an attached network secret
[source,yaml]
----
apiVersion: metal3.io/v1alpha1
kind: BareMetalHost
spec:
# ...
  preprovisioningNetworkDataName: ostest-extraworker-0-network-config-secret
# ...
----

> . To mange the `BareMetalHost` object within the `openshift-machine-api` namespace of your cluster, enter the following command:

Reviewer: Minor nit, maybe more specifically/accurately:
```suggestion
. To mange the `BareMetalHost` object within the `openshift-machine-api` namespace of your cluster, change to the namespace by entering the following command:
```
Since this command itself isn't actually "managing the `BareMetalHost` object" technically. Tho I'm splitting hairs. 🙃

Author: :heavy_check_mark:

+
[source,terminal]
----
$ oc project openshift-machine-api
----

. Get the machine sets:
+
[source,terminal]
----
$ oc get machinesets
----

. Scale each machine set by entering the following command. You must run this command for each machine set.
+
[source,terminal]
----

> $ . oc scale machineset <machineset_name> --replicas=<n> <1>

Reviewer: Stray `.`?
```suggestion
$ oc scale machineset <machineset_name> --replicas=<n> <1>
```

Author: :heavy_check_mark:
